sw/cheri/tests/plic_tests.hh

```diff
@@ -73,10 +80,12 @@ struct PlicTest {
 
       // This lambda will handle the uart specific register to clear the irq.
       irq_handler = [](PlicTest *plic_test, PLIC::Interrupts irq) {
-        auto uart = uart_ptr(plic_test->root, plic_test->instance);
-        plic_test->log(irq, uart->interruptEnable & uart->interruptState);
-        plic_test->error_count += (irq != plic_test->plic_irq_id);
-        plic_test->error_count += !(uart->interruptState & plic_test->ip_irq_id);
+        auto uart             = uart_ptr(plic_test->root, plic_test->instance);
+        const bool wrongIrq   = (irq != plic_test->plic_irq_id);
```

Comment:
Changed. Also made the `log` function take a `failed` argument instead of a `passed` argument to reduce the amount of logical operations being performed here (`!wrong_irq && !wrong_local` -> `wrong_irq || wrong_local` and `!passed` to `failed`).